From 278df41a36fa2eb985007735d5aba7cfff7ac07e Mon Sep 17 00:00:00 2001
From: wang, biao <biao.wang@intel.com>
Date: Wed, 12 Dec 2012 17:10:30 +0800
Subject: [PORT FROM MAIN] Avoid removing audio command that some thread is waiting for status

BZ: 73035

When a client thread insert a comand to mAudioCommands, it try to remove
the same pending command to avoid reduplicative handling, but for the
command with true mWaitStatus, there was a thread waitting for it's
response, removing it would lead this thread never wakeup and lead into
anr.

Change-Id: I038464b888b0dcf91ade91cd6ac103be2348c487
Orig-Change-Id: Ief20e7cc26e25b840d75d152ea020cbd570e153a
Signed-off-by: wang, biao <biao.wang@intel.com>
Reviewed-on: http://android.intel.com:8080/80975
Reviewed-by: Li, Fei <fei.li@intel.com>
Reviewed-by: Kandasamy, Muthukumar <muthukumar.kandasamy@intel.com>
Reviewed-by: Yang, Bin Y <bin.y.yang@intel.com>
Reviewed-by: Nallasellan, Singaravelan <singaravelan.nallasellan@intel.com>
Reviewed-by: Sikkandar D, Madar <madar.sikkandar.d@intel.com>
Tested-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Reviewed-by: P C, SreekanthX <sreekanthx.p.c@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 services/audioflinger/AudioPolicyService.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/AudioPolicyService.cpp b/services/audioflinger/AudioPolicyService.cpp
index 4256fc4..59f0174 100644
--- a/services/audioflinger/AudioPolicyService.cpp
+++ b/services/audioflinger/AudioPolicyService.cpp
@@ -908,6 +908,7 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(AudioCommand *comma
         // commands are sorted by increasing time stamp: no need to scan the rest of mAudioCommands
         if (command2->mTime <= command->mTime) break;
         if (command2->mCommand != command->mCommand) continue;
+        if (command2->mWaitStatus) continue;
 
         switch (command->mCommand) {
         case SET_PARAMETERS: {
-- 
1.7.4.1

